torch/_inductor/utils.py

```diff
@@ -1258,3 +1258,15 @@ def pass_execution_and_save(func, gm, msg):
             t,
             time_elapsed,
         )
+
+
+def is_collective(node):
+    from . import ir
+
+    return isinstance(node, ir.CollectiveKernel) or type(node) == ir._CollectiveKernel
```

Comment:
this is subtle, something related to new types of collectives being lowered differently?

Reply:
You mean `type(node) == ir._CollectiveKernel`? In the new IRs, `ir._WaitKernel` derives from `ir._CollectiveKernel` to share some stuff needed for the cpp codegen.